src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectOptionsProcessor.cs

```diff
@@ -57,45 +58,49 @@ internal class VisualStudioProjectOptionsProcessor : IDisposable
             // Set up _commandLineArgumentsForCommandLine to a default. No lock taken since we're in
             // the constructor so nothing can race.
 
-            // Silence NRT warning.  This will be initialized by the call below to ReparseCommandLineIfChanged_NoLock.
+            // Silence NRT warning.  These will be initialized by the call below to ReparseCommandLineIfChanged_NoLock.
             _commandLineArgumentsForCommandLine = null!;
-            ReparseCommandLineIfChanged_NoLock(commandLine: "");
+            _commandLineStorage = null!;
+            _commandLineChecksum = null!;
+            ReparseCommandLineIfChanged_NoLock(arguments: Enumerable.Empty<string>());
         }
 
         /// <returns><see langword="true"/> if the command line was updated.</returns>
-        private bool ReparseCommandLineIfChanged_NoLock(string commandLine)
+        private bool ReparseCommandLineIfChanged_NoLock(IEnumerable<string> arguments)
         {
-            var checksum = Checksum.Create(commandLine);
+            var checksum = Checksum.Create(arguments);
             if (_commandLineChecksum == checksum)
                 return false;
 
             _commandLineChecksum = checksum;
 
             // Dispose the existing stored command-line and then persist the new one so we can
-            // recover it later.  Only bother persisting things if we have a non-empty string.
+            // recover it later.
+            _commandLineStorage.Dispose();
```

Comment:
i don't see hwo this wouldn't immediately crash.  the constructor sets thsi to null, then calls this method without anyone initializing this value.  So i'm super lost on this :)

Reply:
The tests are a failing right now with null refs, so might be this. @davkean was just asking for high-level review at this point.

Reply:
ah, oik :)  sanity resumes.

Reply:
This isn't tested at all, I've just put it out for a +1 on the approach.